hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,822 @@
+        ax.scale = scale
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            component.centre.value = ref[i]
+        print "Scale changed from  %lf to %lf" % (scale_old, scale)
+
+    def _twin_xray_lines_offset(self, xray_lines):
+        """
+        Twin the offset of the peaks
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        ref = []
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            if i == 0:
+                component_ref = component
+                component_ref.centre.free = True
+                E_ref = component_ref.centre.value
+                ref.append(E_ref)
+            else:
+                component.centre.free = True
+                E = component.centre.value
+                diff = E_ref - E
+                component.centre.twin_function = _get_offset(E, -diff)
+                component.centre.twin_inverse_function = _get_offset(E, diff)
+                component.centre.twin = component_ref.centre
+                ref.append(E)
+        return ref
+
+    def _set_energy_offset(self, xray_lines, ref):
+        """
+        Adjust the width of all lines and set the fitted energy resolution
+        to the spectrum
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        ref: float
+            the centre of the first line before the fit
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        diff = self[xray_lines[0]].centre.value - ref[0]
+        offset_old = self.spectrum.axes_manager[-1].offset
+        self.spectrum.axes_manager[-1].offset -= diff
+        offset = self.spectrum.axes_manager[-1].offset
+        print "Offset changed from  %lf to %lf" % (offset_old, offset)
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            component.centre.value = ref[i]
+
+    def calibrate_energy_axis(self,
+                              calibrate='resolution',
+                              xray_lines='all_alpha',
+                              spread_to_all_lines=True,
+                              **kwargs):
+        """
+        Calibrate the resolution, the scale or the offset of the energy axis
+        by fitting.
+
+        Parameters
+        ----------
+        calibrate: 'resolution' or 'scale' or 'offset'
+            If 'resolution', calibrate the width of all Gaussian. The width is
+            given by a model of the detector resolution, obtained by
+            extrapolation the `energy_resolution_MnKa` in `metadata`
```

Comment:
Is this really accurate? Either way the docstring should specify that this will update the value of `energy_resolution_MnKa`.